Dropdown list (Mobile): "Max width" example and implementation

grzegorz-czajkowski opened the pull request:
Two commits:
 *  UIComponents (Mobile): add "Max width" dropdown example
 *  Dropdown list (Mobile): fit to viewport

grzegorz-czajkowski committed:
UIComponents (Mobile): add "Max width" dropdown example

[Issue] https://github.com/Samsung/TAU/issues/773
[Problem] Missing Dropdown list (Max width) example
[Solution] Add example

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

grzegorz-czajkowski committed:
Dropdown list (Mobile): fit to viewport

[Issue] https://github.com/Samsung/TAU/issues/773
[Problem] Max width example does not fit to view-port
[Solution]
	* apply max-width as 100% view port
	* apply max-height as 100% view port minus margins
	* apply top and bottom paddings instead of line-height as per guide
	* apply padding-right equal to selected icon size (20) plus its
	  paddings (16 + 24)

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

TomaszLukawskiSam commented on src/css/profile/mobile/base.less:
```diff
@@ -76,6 +76,8 @@
 @content-area-line-color: #d6d6d6;
+
+@dropdown-list-margin: 6 * @px_base;
 @dropdown-list-box-shadow-color: fade(@_black, 35%);
```
Try to find and remove this style:
![image](https://user-images.githubusercontent.com/29534410/77542166-0bcf8f80-6ea6-11ea-8ea9-6f37752442ed.png)

grzegorz-czajkowski replied:
Done.

TomaszLukawskiSam commented on src/css/profile/mobile/common/dropdownmenu.less:
```diff
@@ -179,9 +179,10 @@
 	&.ui-dropdownmenu-options-vertical-margins {
-		margin-top: 6 * @px_base;
-		margin-bottom: 6 * @px_base;
+		margin-top: @dropdown-list-margin;
```
do you need this variable for something else? If not, you can remove it.

grzegorz-czajkowski replied:
Yes, I am using it to set container max-height.

TomaszLukawskiSam commented on src/css/profile/mobile/common/dropdownmenu.less:
```diff
@@ -213,6 +214,8 @@
 		list-style: none;
 		padding: 0;
 		margin: 0;
+		max-height: calc(~"100vh - "2 * @dropdown-list-margin);
```
I think that better use simple `2 * 6 * @px_base` instead of additional variables.

grzegorz-czajkowski replied:
Wouldn't be better to have variable for this? So we will have one place to change margin and after this container position and height are updated as well?

TomaszLukawskiSam replied:
I understand you point of view but you need search this value across the project now. In my opinion, if you want to leave this variable you should define it in this file eg. on beginning.

grzegorz-czajkowski replied:
Ok, moved to the beginning of this file.

TomaszLukawskiSam commented on src/css/profile/mobile/theme-changeable/theme.color.less:
```diff
@@ -265,7 +265,6 @@
 //						DropdownMenu
 //***************************************************************************
 @color_dropdownmenu_list_group_stroke: B0733L1; // #[color] list stroke line color
-@color_dropdownmenu_list_item_bg_press: B042; // #[color] list item background press color
```
I think I did mistake in comment.
You can remove this line and replace this variable in:
![image](https://user-images.githubusercontent.com/29534410/77621893-96170280-6f3d-11ea-94cf-981cf283ecfc.png)
by `var(--ripple-color)`

TomaszLukawskiSam replied:
But I don't sure if the press effect is on dropdown menu items.

grzegorz-czajkowski replied:
Yes, @lslachciak noticed that ripple-effect is missing. However, proposed change is not sufficient to have ripple-effect working - it does not disappear. There is a separate issue https://github.com/Samsung/TAU/issues/832 for this and let's take care of this separately.

TomaszLukawskiSam commented on src/css/profile/mobile/common/dropdownmenu.less:
```diff
@@ -213,6 +214,8 @@
+		overflow-y: auto;
 		background-color: var(--dropdown-menu-options-background);
 		border-radius: 26 * @px_base;
 		box-shadow: inset 0 -3 * @px_base 0 0 @dropdown-list-box-shadow-color;
```
My comment is out of scope this change but I think that the `box-shadow` is different as in guideline. `0px 0px 3px 0px rgba(0,0,0,35)`

grzegorz-czajkowski replied:
css box-shadow has different kind of two first parameters (not including inset option). They are (x,y) positions of shadow box not size of border in px. That's why we can not apply them one by one.
I assume the guide intention of
`Box-shadow: 0px 0px 3px 0px rgba(0,0,0,35)`
is to add shadow on each edge, probably starting clockwise.

TomaszLukawskiSam replied:
https://developer.mozilla.org/pl/docs/Web/CSS/box-shadow

grzegorz-czajkowski replied:
Ok will try this.

grzegorz-czajkowski replied:
I applied this change, only rounded borders are having shadow.. Added separate commit fixing this. Thank you.

grzegorz-czajkowski committed:
Dropdown list (Mobile): fix box-shadow

[Issue] N/A
[Problem] Diffirent box-shadow values
	according to guide.
[Solution] Update box-shadow values.

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

TomaszLukawskiSam reviewed (approved):
Ok. Looks good to me.
